index.js

```diff
@@ -47,87 +47,38 @@ Limiter.prototype.inspect = function () {
  * Get values and header / status code and invoke `fn(err, info)`.
  *
  * redis is populated with the following keys
- * that expire after N seconds:
+ * that expire after N milliseconds:
  *
- *  - limit:<id>:count
- *  - limit:<id>:limit
- *  - limit:<id>:reset
+ *  - limit:<id>
  *
  * @param {Function} fn
  * @api public
  */
 
 Limiter.prototype.get = function (fn) {
-  var count = this.prefix + 'count';
-  var limit = this.prefix + 'limit';
-  var reset = this.prefix + 'reset';
+  var key = this.key;
   var duration = this.duration;
   var max = this.max;
   var db = this.db;
-
-  function create() {
-    var ex = (Date.now() + duration) / 1000 | 0;
-
-    db.multi()
-      .set([count, max, 'PX', duration, 'NX'])
-      .set([limit, max, 'PX', duration, 'NX'])
-      .set([reset, ex, 'PX', duration, 'NX'])
-      .exec(function (err, res) {
-        if (err) return fn(err);
-
-        // If the request has failed, it means the values already
-        // exist in which case we need to get the latest values.
-        if (isFirstReplyNull(res)) return mget();
-
-        fn(null, {
-          total: max,
-          remaining: max,
-          reset: ex
-        });
-      });
-  }
-
-  function decr(res) {
-    var n = ~~res[0];
-    var max = ~~res[1];
-    var ex = ~~res[2];
-    var dateNow = Date.now();
-
-    if (n <= 0) return done();
-
-    function done() {
-      fn(null, {
-        total: max,
-        remaining: n < 0 ? 0 : n,
-        reset: ex
-      });
-    }
-
-    db.multi()
-      .set([count, n - 1, 'PX', ex * 1000 - dateNow, 'XX'])
-      .pexpire([limit, ex * 1000 - dateNow])
-      .pexpire([reset, ex * 1000 - dateNow])
-      .exec(function (err, res) {
-        if (err) return fn(err);
-        if (isFirstReplyNull(res)) return mget();
-        n = n - 1;
-        done();
-      });
-  }
-
-  function mget() {
-    db.watch([count], function (err) {
+  var now = Date.now() * 1000 + Math.floor(process.hrtime()[1] / 1000) % 1000;
+  var start = now - duration * 1000;
```

Comment:
Should be `(duration + 1)` to take into account jitter from `hrtime` microseconds.